apps/smtp/src/pages/api/webhooks/invoice-sent.ts

```diff
@@ -84,14 +86,44 @@ const handler: NextWebhookApiHandler<InvoiceSentWebhookPayloadFragment> = async
 
   const useCase = useCaseFactory.createFromAuthData(authData);
 
-  await useCase.sendEventMessages({
-    channelSlug: channel,
-    event: "INVOICE_SENT",
-    payload: { order: payload.order },
-    recipientEmail,
-  });
-
-  return res.status(200).json({ message: "The event has been handled" });
+  return useCase
+    .sendEventMessages({
+      channelSlug: channel,
+      event: "INVOICE_SENT",
+      payload: { order: payload.order },
+      recipientEmail,
+    })
+    .then((result) =>
+      result.match(
+        (r) => {
+          logger.info("Successfully sent email(s)");
+
+          return res.status(200).json({ message: "The event has been handled" });
+        },
+        (err) => {
+          const errorInstance = err[0];
+
+          if (errorInstance instanceof SendEventMessagesUseCase.ServerError) {
+            logger.error("Failed to send email(s) [server error]", { error: err });
+
+            return res.status(500).json({ message: "Failed to send email" });
+          } else if (errorInstance instanceof SendEventMessagesUseCase.ClientError) {
+            logger.info("Failed to send email(s) [client error]", { error: err });
+
+            return res.status(400).json({ message: "Failed to send email" });
+          } else if (errorInstance instanceof SendEventMessagesUseCase.NoOpError) {
+            logger.error("Sending emails aborted [no op]", { error: err });
```

Comment:
question: Should we actually log error here? NoOp error is when settings are missing or config is not active. We don't really have control over this.

```ts
  static NoOpError = BaseError.subclass("SendEventMessagesUseCaseServerError");

  static EventConfigNotActiveError = this.NoOpError.subclass("EventConfigNotActiveError");

  static EventSettingsMissingError = this.NoOpError.subclass("EventSettingsMissingError");
```

Reply:
Yes, you are right. This should be normal log 👍